cli/tests/test_advance_branches.rs

```diff
@@ -0,0 +1,236 @@
+use std::path::Path;
+
+use itertools::Itertools;
+
+use crate::common::TestEnvironment;
+
+fn get_log_output_with_branches(test_env: &TestEnvironment, cwd: &Path) -> String {
+    let template = r#"commit_id.short() ++ " br:{" ++ local_branches ++ "} dsc: " ++ description"#;
+    test_env.jj_cmd_success(cwd, &["log", "-T", template])
+}
+
+fn set_advance_branches(test_env: &TestEnvironment, cwd: &Path, value: bool) -> String {
+    test_env.jj_cmd_success(
+        cwd,
+        &[
+            "config",
+            "set",
+            "--repo",
+            "experimental-advance-branches.enabled",
+            &format!("{}", value),
+        ],
+    )
+}
+
+fn set_advance_branches_overrides(
+    test_env: &TestEnvironment,
+    cwd: &Path,
+    overrides: &[&str],
+) -> String {
+    let override_string: String = overrides.iter().map(|x| format!("\"{}\"", x)).join(",");
+    test_env.jj_cmd_success(
+        cwd,
+        &[
+            "config",
+            "set",
+            "--repo",
+            "experimental-advance-branches.overrides",
+            &format!("[{}]", override_string),
+        ],
+    )
+}
```

Comment:
nit: you can use `test_env.add_config()` (to enable it globally) or `--config-toml` (temporarily).

Reply:
Thanks, I wasn't aware of these options. `add_configs` looks like it will work, but I'm going to make the change in the second config. If it make it in the first config then all of commits will have new change ids and the snapshots will break. This will cause a lot of annoying merge conflicts in the second commit when the tests are parameterized.